src/Plugins/DBFTPlugin/Consensus/ConsensusContext.Get.cs

```diff
@@ -81,18 +102,18 @@ public UInt160 GetSender(int index)
         /// <summary>
         /// Return the expected block size
         /// </summary>
-        public int GetExpectedBlockSize()
+        public int GetExpectedBlockSize(uint pId)
         {
-            return GetExpectedBlockSizeWithoutTransactions(Transactions.Count) + // Base size
-                Transactions.Values.Sum(u => u.Size);   // Sum Txs
+            return GetExpectedBlockSizeWithoutTransactions(Transactions[pId].Count) + // Base size
+                Transactions[pId].Values.Sum(u => u.Size);   // Sum Txs
         }
 
         /// <summary>
         /// Return the expected block system fee
         /// </summary>
-        public long GetExpectedBlockSystemFee()
+        public long GetExpectedBlockSystemFee(uint pId)
         {
-            return Transactions.Values.Sum(u => u.SystemFee);  // Sum Txs
+            return Transactions[pId].Values.Sum(u => u.SystemFee);  // Sum Txs
```

Comment:
I think you should add some kind of error `TryCatch` to catch if `KeyNotFound` or `InvalidIndex` on  `pId`

Reply:
Can you create a PR trying to improve that? Pointing to this branch?

Reply:
I can, will be later though.

Reply:
ping @ZhangTao1596

Reply:
What should the `value` be when it errors?

Reply:
let me check.


Any change in this PR should be done with another PR, for the sake of organization and security.

Reply:
I check here. Usually, pId will always have value, @cschuchardt88.

It is a obligatory parameter uint of the function.

Reply:
https://github.com/neo-project/neo/blob/20d0b49cedb85c6d61b4a0db691d8e4be52ee0ed/src/Plugins/DBFTPlugin/Consensus/ConsensusContext.cs#L303

Reply:
Lets say this method crashes cause of `IndexOutOfRangeException` than what?

Reply:
`Transactions[pID] = transactions.Length == 0 && !RequestSentOrReceived ? null : transactions.ToDictionary(p => p.Hash);`

Reply:
This loop here https://github.com/neo-project/neo/blob/20d0b49cedb85c6d61b4a0db691d8e4be52ee0ed/src/Plugins/DBFTPlugin/Consensus/ConsensusContext.cs#L320-L347